### PR TITLE
DAOS-623 engine: Fix a typo

### PR DESCRIPTION
**Pull request by jolivier23**
tgt_nr should divide by tgt_per_numa, not number of numa nodes.  Otherwise, it can trigger the assertion at the end of the function.

Required-githooks: true

Change-Id: I92f65924f9b4b3dce6a756e01e5bfc9e584af0b6

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by NiuYawei — APPROVED** [review]

File: src/engine/ult.c [hunk 1]
```diff
@@ -374,30 +374,27 @@ static inline uint32_t
...
 		}
 		return DSS_XS_SELF;
 	}
 
-	socket  = tgt_id / dss_numa_nr;
-	base    = dss_sys_xs_nr + dss_tgt_nr + (socket * dss_offload_per_numa_nr);
-	target  = base + ((offload + tgt_id) % dss_offload_per_numa_nr);
-	offload = target + 17; /* Seed next selection */
+	base   = dss_sys_xs_nr + dss_tgt_nr + (socket * dss_offload_per_numa_nr);
+	target = base + ((offload + tgt_id) % dss_offload_per_numa_nr);
```
Comment (Nasf-Fan) [inline src/engine/ult.c:393]: Do we need to keep the "target" and "tgt_id" within the same NUMA node or not? If yes, how? It seems that "base" may exceeds the boundary?

Comment (jolivier23) [reply src/engine/ult.c:393]: Target in this calculation is a helper thread and will be on the same numa node according to the expected layout shown here: 

image

Comment (jolivier23) [reply src/engine/ult.c:393]: To be more precise, the helper threads are split among numa nodes but base is the xs id of the first helper thread.

Comment (Nasf-Fan) [reply src/engine/ult.c:393]: Thank for the detailed explanation.

**Review by Nasf-Fan — APPROVED** [review]

**Pull request merged**